src/main/java/com/github/phantomthief/collection/impl/GenericSimpleBufferTriggerBuilder.java

```diff
@@ -58,11 +59,30 @@
         return this;
     }
 
+    public GenericSimpleBufferTriggerBuilder<E, C> tickTime(long time, TimeUnit unit) {
+        builder.tickTime(time, unit);
+        return this;
+    }
+
+    public GenericSimpleBufferTriggerBuilder<E, C>
+            triggerStrategy(TriggerStrategy triggerStrategy) {
+        builder.triggerStrategy(triggerStrategy);
+        return this;
+    }
+
+    /**
+     * use {@link #interval(long, TimeUnit)} or {@link #triggerStrategy}
+     */
+    @Deprecated
     public GenericSimpleBufferTriggerBuilder<E, C> on(long interval, TimeUnit unit, long count) {
         builder.on(interval, unit, count);
         return this;
     }
 
+    public GenericSimpleBufferTriggerBuilder<E, C> interval(long interval, TimeUnit unit) {
```

Comment:
ok. it should check invalid trigger settings and throw exception while building the trigger.